src/vm/checker/typecheck/natives/mod.rs

```diff
@@ -153,13 +153,76 @@ fn check_special_let(checker: &mut TypeChecker, args: &[SymbolicExpression], con
     let binding_list = args[0].match_list()
         .ok_or(CheckError::new(CheckErrors::BadLetSyntax))?;
     
-    let let_context = checker.type_check_list_pairs(binding_list, context)?;
```

Comment:
We could de-duplicate some of this code by adding a boolean parameter to `type_check_list_pairs` that was like `check_names_unique`.

Reply:
Right.
I initially started handling #1021 in this PR, and this function was (and will) also responsible for handling this:
```
        (define (set-cursor (value int))
            (let ((cursor 1) (cursor 2))
               value))
```

By having this, in the loop:
```
        if checker.constant_types.contains_key(var_name) {
            Err(CheckError::new(CheckErrors::NameAlreadyUsed(name.to_string())))
        }
```
- undesired in a generic type_check_list_pairs.

Reply:
In any case, I think this was the only usage of `type_check_list_pairs` -- so maybe just remove that function from the checker impl?

Reply:
Addressed with https://github.com/blockstack/blockstack-core/pull/1022/commits/908fd898f3b1d83dd3b12d9d6a7ed2886a8c77da